.changeset/cold-schools-yell.md

```diff
@@ -0,0 +1,34 @@
+---
+'astro': minor
+---
+
+Add integrations dynamically
+
+Astro integrations can now dynamically add additional integrations during set-up. This makes it possible for integration authors to bundle integrations more intelligently for their users.
```

Comment:
```suggestion
Astro integrations can now themselves dynamically add and configure additional integrations during set-up. This makes it possible for integration authors to bundle integrations more intelligently for their users.
``` 
I just kind of like this for emphasis, since we have some level of recursion going on?

And just checking that if you only got one phrase to describe this feature, it would be "intelligent bundling" that's the key idea here, and not for example something about ensuring dependencies because now it can know what other integrations you have?

Reply:
Bit biased because I’m thinking about Starlight being able to switch from always adding MDX & Sitemap to only adding them when needed — this is what I had in mind for “intelligent bundling”. I think that makes sense seeing as technically integrations could provide their dependencies (like Starlight does today), but not “intelligently” — e.g. if a user adds sitemap themselves, things crash.